File: cpp/tree/invert_tree.cpp
#include <bits/stdc++.h>
using namespace std;

#define ll long long
#define pb push_back
#define all(x) x.begin(), x.end()
#define endl '\n'

struct TreeNode {
    int val;
    TreeNode* left;
    TreeNode* right;
    TreeNode()
      : val(0)
      , left(nullptr)
      , right(nullptr) {
    }
    TreeNode(int x)
      : val(x)
      , left(nullptr)
      , right(nullptr) {
    }
    TreeNode(int x, TreeNode* left, TreeNode* right)
      : val(x)
      , left(left)
      , right(right) {
    }
};

TreeNode* invertTree(TreeNode* root) {

    return nullptr;
}

int main() {

    ios::sync_with_stdio(0);
    cin.tie(0);

    return 0;
}

File: python/tree/invert_tree.py
import sys
import threading
from typing import List, Tuple, Dict, Set
from collections import defaultdict, deque


class Solution:

    class TreeNode:
        def __init__(self, val=0, left=None, right=None):
            self.val = val
            self.left = left
            self.right = right

    def invertTree(self, root):
        if root is None:
            return None

        root.left, root.right = root.right, root.left
        self.invertTree(root.left)
        self.invertTree(root.right)

        return root

    def invertTree(self, root):
        if not root:
            return None

        stack = [root]

        while stack:
            node = stack.pop()
            node.left, node.right = node.right, node.left
            if node.left:
                stack.append(node.left)
            if node.right:
                stack.append(node.right)

        return root


def main():
    s = Solution()
    data = [1, 2, 3]


if __name__ == "__main__":
    threading.Thread(target=main).start()
